pyupgrade.py

```diff
@@ -791,9 +850,14 @@ def _fix_percent_format_tuple(tokens, start, node):
 
 
 def _fix_percent_format_dict(tokens, start, node):
-    seen_keys = set()
+    # type: (List[Token], int, ast.BinOp) -> None
+    seen_keys = set()  # type: Set[Text]
     keys = {}
-    for k in node.right.keys:
+
+    # the caller has enforced this, but we have to tell mypy again
+    right = cast(ast.Dict, node.right)
```

Comment:
I usually prefer `assert` instead of `cast` -- it has the same affect but with the added benefit of being checked

Reply:
sure